src/core/components/table/AttendanceTable.vue

```diff
@@ -0,0 +1,86 @@
+<template>
+  <q-card>
+    <q-table :data="course.trainees" :columns="columns" class="q-pa-md" hide-bottom>
+      <template #header="props">
+        <q-tr :props="props">
+          <q-th v-for="col in props.cols" :key="col.name" :props="props">
+            <div v-if="!['traineeName','traineePicture'].includes(col.name)">
+              <div class="text-primary date">{{ col.month }}</div>
+              <div class="number">{{ col.day }}</div>
+              <div class="text-weight-bold date">{{ col.weekDay }}</div>
+              <div class="text-grey">
+                <div>{{ col.startHour }}</div>
+                <div>{{ col.endHour }}</div>
+              </div>
+              <div class="text-grey-800">
+                <q-icon name="supervisor_account" />
+                {{ col.trainees }}
+              </div>
+            </div>
+          </q-th>
+        </q-tr>
+      </template>
+      <template #body="props">
+        <q-tr :props="props">
+          <q-td v-for="col in props.cols" :key="col.name" :props="props">
+            <img v-if="col.name === 'traineePicture'" class="q-mx-md avatar"
+              :src="col.value ? col.value.link : DEFAULT_AVATAR">
+            <div v-else-if="col.name === 'traineeName'">
+              {{ formatIdentity(col.value, 'FL') }}
+            </div>
+            <q-checkbox v-else :value="false" dense size="sm" />
+          </q-td>
+        </q-tr>
+      </template>
+    </q-table>
+    <ni-button class="q-my-sm" icon="add" color="primary" label="Ajouter un participant" />
+  </q-card>
+</template>
+
+<script>
+import moment from '@helpers/moment';
+import Button from '@components/Button';
+import { upperCaseFirstLetter, formatIdentity } from '@helpers/utils';
+import { DEFAULT_AVATAR } from '@data/constants';
+
+export default {
+  name: 'ExpandingTable',
+  props: {
+    course: { type: Object, default: () => ({}) },
+  },
+  components: {
+    'ni-button': Button,
+  },
+  data () {
+    return {
+      formatIdentity,
+      DEFAULT_AVATAR,
+    };
+  },
+  computed: {
+    columns () {
+      const columns = this.course.slots.map(s => ({
+        name: s._id,
```

Comment:
Tu pourrais mettre un nom avec une racine commune (ex: `slot-${s._id}) ca permettrait de faire le v-if dans le v-for en positif plutot qu'en negatif si on ajoute d'autre colonne que traineeName ou traineePicture ou si on change leur nom, il n'y aura pas de soucis pour les colonnes de creneaux.

Reply:
Il me semble que l'_id du slot est egalement dans value non ? (comme dans field tu mets _id?

Reply:
l' _id de field correspond à la clé dans mon element du tableau data que je souhaite afficher dans cette colonne, donc l'_id dans col.value est celui de l'apprenant